magneticfield/inc/fieldPropagatorConstBz.h

```diff
@@ -69,33 +69,33 @@ __host__ __device__ double fieldPropagatorConstBz::ComputeStepAndPropagatedState
     vecgeom::NavStateIndex &next_state)
 {
   double momentumMag = sqrt(kinE * (kinE + 2.0 * mass));
+  double momentumXYMag =
+      momentumMag * sqrt((1. - direction[2]) * (1. + direction[2])); // only XY component matters for the curvature
 
-  double curv   = std::fabs(ConstBzFieldStepper::kB2C * charge * BzValue) / (momentumMag + 1.0e-30); // norm for step
+  double curv = std::fabs(ConstBzFieldStepper::kB2C * charge * BzValue) / (momentumXYMag + 1.0e-30); // norm for step
```

Comment:
The safe length to progress in the plane will be proportional to momentumXYMag. But when we use it in 3-d it should become proportional to momentumMag.